nupic/research/connections.py

```diff
@@ -238,6 +238,105 @@ def numSynapses(self):
     return len(self._synapses)
 
 
+  def write(self, proto):
+    protoCells = proto.init('cells', self.numCells)
+
+    for cell in xrange(self.numCells):
+      segments = self.segmentsForCell(cell)
+      protoSegments = protoCells[cell].init('segments', len(segments))
+
+      for j, segment in enumerate(segments):
+        synapses = self.synapsesForSegment(segment)
+        protoSynapses = protoSegments[j].init('synapses', len(synapses))
+
+        for k, synapse in enumerate(synapses):
+          synapseData = self.dataForSynapse(synapse)
+          protoSynapse = protoSynapses[k]
+
+          protoSynapse.presynapticCell = synapseData.presynapticCell
+          protoSynapse.permanence = synapseData.permanence
+
+
+  @classmethod
+  def read(cls, proto):
+    protoCells = proto.cells
+    connections = object.__new__(cls)
+    connections.__init__(len(protoCells))
```

Comment:
If you want to call `__init__` then just replace the `__new__` and `__init__` calls with a regular instantiation of the class. That said, we have usually used `__new__` to avoid the initialization that happens in `__init__` since it is usually redundant to the loading from the proto object.

So I'd say it is probably better to avoid `__init__` unless it contains logic that is necessary for initialization even when loading from a checkpoint, in which case just do `connections = cls(len(protoCells))` or similar. Also, if you keep the `__new__` call then it should be `connections = cls.__new__()`